server/api/cards.go

```diff
@@ -0,0 +1,388 @@
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+	userID := getUserID(r)
+	boardID := mux.Vars(r)["boardID"]
+
+	query := r.URL.Query()
+	strPage := query.Get("page")
+	strPerPage := query.Get("per_page")
+
+	if !a.permissions.HasPermissionToBoard(userID, boardID, model.PermissionViewBoard) {
+		a.errorResponse(w, r.URL.Path, http.StatusForbidden, "", PermissionError{"access denied to fetch cards"})
+		return
+	}
+
+	if strPage == "" {
+		strPage = defaultPage
+	}
+	if strPerPage == "" {
+		strPerPage = defaultPerPage
+	}
+
+	page, err := strconv.Atoi(strPage)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusBadRequest, "invalid `page` parameter", err)
+	}
+
+	perPage, err := strconv.Atoi(strPerPage)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusBadRequest, "invalid `per_page` parameter", err)
+	}
+
+	auditRec := a.makeAuditRecord(r, "getCards", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelRead, auditRec)
+	auditRec.AddMeta("boardID", boardID)
+	auditRec.AddMeta("page", page)
+	auditRec.AddMeta("per_page", perPage)
+
+	cards, err := a.app.GetCardsForBoard(boardID, page, perPage)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	a.logger.Debug("GetCards",
+		mlog.String("boardID", boardID),
+		mlog.Int("page", page),
+		mlog.Int("per_page", perPage),
+		mlog.Int("count", len(cards)),
+	)
+
+	data, err := json.Marshal(cards)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	// response
+	jsonBytesResponse(w, http.StatusOK, data)
+
+	auditRec.Success()
+}
+
+func (a *API) handlePatchCard(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation PATCH /cards/{cardID}/cards patchCard
+	//
+	// Patches the specified card.
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: cardID
+	//   in: path
+	//   description: Card ID
+	//   required: true
+	//   type: string
+	// - name: Body
+	//   in: body
+	//   description: the card patch
+	//   required: true
+	//   schema:
+	//     "$ref": "#/definitions/CardPatch"
+	// - name: disable_notify
+	//   in: query
+	//   description: Disables notifications (for bulk data patching)
+	//   required: false
+	//   type: bool
+	//  security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       $ref: '#/definitions/Card'
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	userID := getUserID(r)
+	cardID := mux.Vars(r)["cardID"]
+
+	val := r.URL.Query().Get("disable_notify")
+	disableNotify := val == True
+
+	requestBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	cardBlock, err := a.app.GetBlockByID(cardID)
```

Comment:
I think we should use `GetCardByID` and have the card related endpoints only interact with `Cards` from the API perspective, and if there is any `Block to Card` conversion, it should happen in the `App` layer

Reply:
Fixed via https://github.com/mattermost/focalboard/pull/3760/commits/65055ac6f2703631b11de467eab458bf3478a3eb